llama_index/llms/konko_utils.py

```diff
@@ -145,15 +158,19 @@ def to_openai_message_dicts(messages: Sequence[ChatMessage]) -> List[dict]:
     return [to_openai_message_dict(message) for message in messages]
 
 
-def from_openai_message_dict(message_dict: dict) -> ChatMessage:
+def from_openai_message_dict(message_dict: Any) -> ChatMessage:
     """Convert openai message dict to generic message."""
-    role = message_dict["role"]
-    content = message_dict.get("content", None)
-
-    additional_kwargs = message_dict.copy()
-    additional_kwargs.pop("role")
-    additional_kwargs.pop("content", None)
-
+    if is_openai_v1():
+        role = message_dict.role
+        content = message_dict.content
+    else:
+        role = message_dict["role"]
+        content = message_dict.get("content", None)
+    additional_kwargs = {
+        attr: getattr(message_dict, attr)
```

Comment:
how does this work with a `dict` returned by the version 0?

Reply:
done